src/CryptoUtils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Helper function that turns a string into a unique 53-bit hash.
+ * @ref https://stackoverflow.com/a/52171480/6456163
+ * @param {string} str
+ * @param {number} seed
+ * @returns {number}
+ */
+export const cyrb53 = (str, seed = 0) => {
+  let h1 = 0xdeadbeef ^ seed, h2 = 0x41c6ce57 ^ seed;
+  for (let i = 0, ch; i < str.length; i++) {
+    ch = str.charCodeAt(i);
+    h1 = Math.imul(h1 ^ ch, 2654435761);
+    h2 = Math.imul(h2 ^ ch, 1597334677);
+  }
+  h1  = Math.imul(h1 ^ (h1 >>> 16), 2246822507);
+  h1 ^= Math.imul(h2 ^ (h2 >>> 13), 3266489909);
+  h2  = Math.imul(h2 ^ (h2 >>> 16), 2246822507);
+  h2 ^= Math.imul(h1 ^ (h1 >>> 13), 3266489909);
+
+  return 4294967296 * (2097151 & h2) + (h1 >>> 0);
+};
```

Comment:
We cannot accept this algo (or derivations thereof) due to its problematic usage rights situation.

Reply:
Would this solution be satisfactory?

https://github.com/bryc/code/issues/23#issuecomment-2051906976

Reply:
Thanks for the effort, but unfortunately the underlying issue is still unaddressed. I appreciate the author's emotional responses when it comes to OS licenses. However, the issue is a legal one, not an emotional one. Unless there's _an official OS license_ attributed to the code, we cannot accept it. We also cannot accept any "self-written", non-official OS license on the same grounds.

Reply:
@mtrezza is the MIT license addition satisfactory?

Reply:
Yes, but SO is not sufficient as source or link to the author. Has this been published anywhere on GitHub, and can this be added as a dependency? Because if we add code like this it won't be maintained.